### DISCUSSION TITLE
Confusion matrix

### DISCUSSION THREAD

**Discussion by rabiyaabbasi** [ultralytics/yolov5#7324] [Q&A]
As I am new to YoloV5, I have some questions related to the confusion matrix that I got after training the data. Fisrt of all, is it normalized confusion matrix? And in first cell the value is 0.50, does it mean that model have predicted correctly 50% samples for the particular class correctly. Secondly, why there are background FN and background FP in my confusion matrix. What do they indicate? 

Regards
![Capture2](https://user-images.githubusercontent.com/54917838/162084144-d8536649-931c-4a4e-8f38-5e571d444734.PNG)
.

**Comment by glenn-jocher** [comment]
@rabiyaabbasi columns are normalized. Predicted classes in each row, True classes in each column. Perfect result would be diagonal of 1.0's. FN and FP are False Negatives and False Positives. See https://en.wikipedia.org/wiki/Precision_and_recall

**Reply by rabiyaabbasi** [reply]
Thank you for your prompt response.  I understand how the confusion matrix works except that I am really confused about the last row and last column. What is their interpretation? For instance, for the second class (fungal-powdery mildew),  what does value of background FN (0.50) and background FP(0.14) interpret?

**Reply by SaddamHosyn** [reply]
hi rabiya. have you got a satisfactory answer for Background FN and Background FP?

**Reply by leandroipca** [reply]
> hi rabiya. have you got a satisfactory answer for Background FN and Background FP?

Hey, i'm trying to find the meaning  Background FN and Background FP....

**Reply by glenn-jocher** [reply]
@leandroipca these terms have been removed in master. If you see them your code is out of date.
